src/Saturn/CSRF.fs

```diff
@@ -0,0 +1,49 @@
+namespace Saturn
+
+module CSRF =
+  open FSharp.Control.Tasks.ContextInsensitive
+  open Giraffe.Core
+  open Giraffe.ResponseWriters
+  open Microsoft.AspNetCore.Antiforgery
+  open Microsoft.AspNetCore.Http
+
+  let private shouldValidate  =
+    let unprotectedMethods = Set.ofList ["GET"; "HEAD"; "TRACE"; "OPTIONS"]
+    fun (ctx: HttpContext) -> not <| unprotectedMethods.Contains ctx.Request.Method
+
+  let inline private sendException (ex: AntiforgeryValidationException) =
+    setStatusCode 403
+    >=> text ex.Message
+
+  /// Protect a resource by validating that requests that can change state come with a valid request antiforgery token, which is based off of a known session token.
+  /// The particular configuration options can be set via the `application` builder's `use_antiforgery_with_config` method.
+  let csrf : HttpHandler =
+    fun (next) (ctx) -> task {
+     if shouldValidate ctx
+     then
+      try
+        do! ctx.GetService<IAntiforgery>().ValidateRequestAsync(ctx)
+        return! next ctx
+      with
+      | :? AntiforgeryValidationException as ex ->
```

Comment:
What will happen if we use this Handler without initializing csrf in `application` CE?

Reply:
the dreaded null-reference. I can add a logging statement reminding the user to add the configuration to the `application` running the app?